packages/synapse-react-client/src/components/QueryVisualizationWrapper.tsx

```diff
@@ -138,8 +139,12 @@ export function QueryVisualizationWrapper(
   } = useQueryContext()
 
   let { rowSelectionPrimaryKey } = props
-  if (!rowSelectionPrimaryKey && isFileViewOrDataset(entity)) {
-    // If the primary key isn't specified on a file view/dataset, we can safely use the 'id' column
+  if (
+    !rowSelectionPrimaryKey &&
+    isFileViewOrDataset(entity) &&
+    data?.columnModels?.find(cm => cm.name === 'id')
+  ) {
+    // If the primary key isn't specified on a file view/dataset, we can safely use the 'id' column, if it is present
```

Comment:
This comment does not make sense to me.  Is there a way to specify the primary key on a file view/dataset?

Reply:
should be changed to read "If the primary key isn't specified via props, and this is a file view or dataset..."